src/main/java/bio/terra/service/dataset/flight/ingest/IngestParseJsonFileStep.java

```diff
@@ -58,18 +53,24 @@ public StepResult doStep(FlightContext flightContext)
     workingMap.put(IngestMapKeys.TABLE_SCHEMA_FILE_COLUMNS, fileRefColumnNames);
 
     List<String> errors = new ArrayList<>();
-    List<JsonNode> fileLineJson =
-        gcsFileLines.stream()
-            .map(
-                content -> {
-                  try {
-                    return objectMapper.readTree(content);
-                  } catch (JsonProcessingException ex) {
-                    errors.add(ex.getMessage());
-                    return null;
-                  }
-                })
-            .collect(Collectors.toList());
+    Set<BulkLoadFileModel> fileModels =
+        IngestUtils.getJsonNodesStreamFromFile(gcsPdao, objectMapper, ingestRequest, dataset)
+            .flatMap(pair -> IngestUtils.resolveJsonNodeCollectError(pair, errors))
```

Comment:
`IngestUtils.getJsonNodesStreamFromFile` returns a `Pair<String, JsonNode>`, which I'm using like a Scala `Either` to collect errors. `IngestUtils.resolveJsonNodeCollectError` collects those errors by modifying a list and adding errors to it. Its just getting rid of nulls.

Reply:
My point is that if you pass `errors` directly to `getJsonNodesStreamFromFile()` then you can add errors to directly instead of having to create a `Pair` result, which is later used to populate `errors`.